Jenkinsfile

```diff
@@ -94,37 +117,58 @@ def publishChart(imageName) {
 
 node {
   checkout scm
-  stage('Set branch, PR, and containerTag variables') {
-    (branch, pr, containerTag, mergedPrNo) = getVariables(repoName)
-  }
-  stage('Build test image') {
-    buildTestImage(testImageName, BUILD_NUMBER)
-  }
-  stage('Run tests') {
-    runTests(testImageName, BUILD_NUMBER)
-  }
-  stage('Build production image') {
-    buildProductionImage(imageName, BUILD_NUMBER)
-  }
-  stage('Push container image') {
-    pushContainerImage(registry, regCredsId, imageName, containerTag)
-  }
-  if (pr != '') {
-    stage('Helm install') {
-      def extraCommands = "--values ./helm/$repoName/jenkins-aws.yaml"
-      deployPR(kubeCredsId, registry, imageName, containerTag, extraCommands)
-      echo "Build available for review"
+
+  try {
+    stage('Set branch, PR, and containerTag variables') {
+      (branch, pr, containerTag, mergedPrNo, repoUrl, commitSha) = getVariables(repoName)
+
+      if (pr) {
+        sh "echo Building $pr"
+      } else if (branch == "master") {
+        sh "echo Building master branch"
+      } else {
+        currentBuild.result = 'ABORTED'
+        error('Build aborted - not a PR or a master branch')
+      }
     }
-  }
-  if (pr == '') {
-    stage('Publish chart') {
-      publishChart(imageName)
+    stage('Report pending status') {
+      updateGithubCommitStatus('Build started', 'PENDING', repoUrl, commitSha)
     }
-  }
-  if (mergedPrNo != '') {
-    stage('Remove merged PR') {
-      sh "echo removing deployment for PR $mergedPrNo"
-      undeployPR(kubeCredsId, imageName, mergedPrNo)
+    stage('Build test image') {
+      buildTestImage(testImageName, BUILD_NUMBER)
+    }
+    stage('Run tests') {
+      runTests(testImageName, BUILD_NUMBER)
+    }
+    stage('Build production image') {
+      buildProductionImage(imageName, BUILD_NUMBER)
+    }
+    stage('Push container image') {
+      pushContainerImage(registry, regCredsId, imageName, containerTag)
+    }
+    if (pr != '') {
+      stage('Helm install') {
+        def extraCommands = "--values ./helm/$repoName/jenkins-aws.yaml"
```

Comment:
The deployment needs to override the service URLs as there are no FFC services available in the `default` namespace. The namespace for master deployments is `ffc-demo` in the AWS cluster.

Reply:
Good spot. In fact, if the services are deployed to the same namespace then we don't need to specify the namespace at all in URLs, which seems a little more flexible. I've tested this on my local K8s cluster using `wget -qO- http://ffc-demo-user-service/healthz` from within an API Gateway container in the same namespace as the user service.

I've made that change in `values.yaml` so that services can communicate regardless of what namespace they're deployed to (as long as they share a namespace, which was the case when specifying `default` anyway).

Reply:
Ah, that's a good idea. Updated.